src/components/tabset/tabset.token.ts

```diff
@@ -0,0 +1,6 @@
+import { InjectionToken } from '@angular/core';
+
+/**
+ * This token is used to avoid circular dependency
+ */
+export const TabsetToken = new InjectionToken('Tabset');
```

Comment:
This is something we should update in Universal, but injection tokens should ideally contain the type information of the thing it will provide. This is not so important when using constructor injection, but as of Angular 14 you can use an `inject` function instead so this provides type inference. E.g.

```ts
readonly tabset = inject(TabsetToken);
```

We will likely be changing our components to use this approach once we drop Angular 13 support as it allows us to more easily extend components as we don't need to forward all constructor parameters to parent component classes.

```suggestion
import type { TabsetComponent } from './tabset.component';

export const TabsetToken = new InjectionToken<TabsetComponent>('Tabset');
```